### ISSUE TITLE
Include OCLC / OCN id option on add book page.

### ISSUE THREAD

**Issue by hornc** [internetarchive/openlibrary#8719]
https://openlibrary.org/books/add

should provide an option to add an OCLC/WorldCat number (OCN) as an identifier.

It is one of the ids available on the edit page. It seems sensible to also include it on the add page.

**Comment by Ayush1404** [comment]
Hey I would like to work on this issue,I believe I can do it

**Comment by Ayush1404** [comment]
Hey I would like to work on this issue,I believe I can do it

**Comment by cdrini** [comment]
@hornc Should we just include the full identifiers list at this point? It might be nice to DRY the rendering and validation code. CC @seabelis 

@Ayush1404 there's still a bit of discussion to be had around this one so not yet ready to work on :)

**Comment by Ayush1404** [comment]
Sure , I would love to work on it once the discussion is done , thanks for the reply :)

**Comment by seabelis** [comment]
"Should we just include the full identifiers list at this point? "

I'd rather not. Only one can be added at a time on that form; I'd prefer to keep it limited to the most important ones and not social or retail sites.

**Comment by hornc** [comment]
Yes, I agree with @seabelis , no need to confuse the options, many of which are hardly used. ISBN, LCCN, and OCLC are the three main identifiers we tend to use, and OCLC number is more used than LCCN in some of our systems.

Anecdotally, OCLC numbers get used quite a bit on WIkipedia when people enter references, so it is a book identifier that online bibliographs are familiar with.

**Comment by Freso** [comment]
> Only one can be added at a time on that form

As I understood @cdrini’s comment, he wanted to include the same identifier “widget” on the add page that’s currently on the edit page (i.e., [reuse that code to reduce overall code complexity](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)). This would by extension also mean being able to add multiple identifiers when adding a book. I, personally, think this would be great.

---------------

If we don’t want to switch to allowing to add multiple identifiers when adding books, then I agree that not all identifiers should be added to this list—but I also then think the current list of options is too sparse.

`isbn_1[03]` should obviously be there.
I get wanting “trusted book providers” (e.g., librivox, gutenberg) too. We’re still missing a few from this list (`wikisource`, `openstax`, `standard_ebooks`, + one more soon… :eyes:), so I’m guessing these ought to be added too?
If OCLC, then ISNI and VIAF might also make sense, being international library system/collaboration identifiers too.
LCCN, on the other hand… I guess? But then we should also, IMHO, add all other national library identifiers, to counter US centricity.

Practical questions:
- [`openlibrary/templates/books/add.html`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/books/add.html) obviously needs editing, but AFAICT no other files would be needed to edit this. Am I missing something?
- the [`$popular` array in `templates/books/edit/edition.html`](https://github.com/internetarchive/openlibrary/blob/e953316784b4c1153d9a799579a91641f5376a08/openlibrary/templates/books/edit/edition.html#L444) is _almost_ identical to the list on the add book page – only thing missing from add book page is OCLC, which, well, is what this issue is (originally) about. :) Should the two be sync’d/kept in sync?

**I’d like to be assigned this** if either
1. the issue remains limited to _just_ OCLC, or
2. I can add remaining trusted identifiers and inter/national library identifiers (ISNI, VIAF, and [whichever library edition identifiers we currently support](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/config/edition/identifiers.yml) (that I can recognise/identify)).

I _do not_ want to be assigned this if you want to balloon or repurpose the issue into one for having a better UI on the add page for adding identifiers. (IMHO, if you want to do this, you should make a new issue for that, and allow a simple PR against `openlibrary/templates/books/add.html` adding OCLC for this issue. Don’t let perfect be the enemy of good and all that.)

**Comment by hornc** [comment]
@Freso I think this issue should be limited to OCLC, so your number 1 above, and I think we can assign this to you.

Good spotting re. the $popular array, it's interesting to see it there, I'm not sure exactly where that is used, and the justification. I'd argue that Librivox and Project Gutenberg ids should be imported more systematically by a process other than user submitted edits, but those are already on the [books/add.html](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/books/add.html) page anyway. I think there might be a "show of solidarity" and boosting similar open projects happening there? They should be left alone for now, but I don't think `$popular` is necessarily the same list as this page.

The use case as I saw it is users adding pre-ISBN books to OL via the wiki interface who wanted to include a unique, disambiguating identifier. Without getting into too much detail, OCLC numbers do seem to be the only available bibliographic id type that can potentially be formally assigned to _any_ book ever published, regardless of where copies are held. 

Between ISBN and OCLC number, _every_ book should be able to have an identifier.  This issue isn't just about saying "more identifiers are better", in fact there are good reasons _not_ clutter the interface. It's about getting identifier coverage on non-ISBN books that has at least the potential to be complete, and is usable in practical bibliographic exchanges. 

I've given examples above how anecdotally it appears to be frequently used this way on Wikipedia for non-ISBN books, and _specifically_ for archive.org integration with Open Library and other library partners, archive.org uses OCLC numbers.
